File: command.js
let Receiver_3 = function () {
    //Empty
};

Receiver_3.prototype = {
    command: function () {
        console.log("Command is triggered!")
    }
};

let Receiver = function (rec_2) {
    this.rec_2 = rec_2;
};

Receiver.prototype = {
    command: function () {
        this.rec_2.command();
    }
};

let Invoker = function (rec) {
    this.rec = rec;
}

Invoker.prototype = {
    command: function () {
        this.rec.command();
    }
};

let main = () => {
    let rec_3 = new Receiver_3();
    let rec_2 = new Receiver(rec_3);
    let rec = new Receiver(rec_2);
    let inv = new Invoker(rec);
    inv.command();
}

main()

File: command.java
public class command{
	static private interface Receivable {
		void command();
	}

	static private class Receiver_3 implements Receivable{
		public void command(){
			System.out.println("Command is triggered!");
		}
	}

	static private class Receiver implements Receivable{
		private Receivable rec = null;
		Receiver(Receivable rec){
			this.rec = rec;
		}

		public void command(){
			this.rec.command();
		}
	}	

	static private class Invoker{
		private Receivable rec = null;
		Invoker(Receivable rec){
			this.rec = rec;
		}

		public void command(){
			this.rec.command();
		}
	}

	public static void main(String[] args) {
		Receivable rec_3 = new Receiver_3();
		Receivable rec_2 = new Receiver(rec_3);
		Receiver rec = new Receiver(rec_2);
		Invoker inv = new Invoker(rec);
		inv.command();
	}
}
